migrations/20240916110123_create_items_table.js

```diff
@@ -0,0 +1,40 @@
+exports.up = function (knex) {
+  return knex.schema.createTable('items', function (table) {
+    table.increments('id').primary();
+    table
+      .integer('owner_id')
+      .unsigned()
+      .notNullable()
+      .references('id')
+      .inTable('users')
+      .onDelete('CASCADE');
+    table
+      .integer('bucket_id')
+      .unsigned()
+      .notNullable()
+      .references('id')
+      .inTable('buckets')
+      .onDelete('CASCADE');
+    table
+      .integer('parent_id')
+      .unsigned()
+      .nullable()
+      .references('id')
+      .inTable('items')
+      .onDelete('SET NULL');
```

Comment:
```suggestion
```

Same here. No auto-deletion. If a parent delete, we should remove all child items.